lib/Mount/CacheRootPermissionsMask.php

```diff
@@ -36,11 +36,15 @@ public function __construct(ICache $cache, int $mask) {
 
 	protected function formatCacheEntry($entry) {
 		$path = $entry['path'];
-		$isRoot = $path === '' || (strpos($path, '__groupfolders') === 0 && count(explode('/', $path)) === 2);
+		$isRoot = $path === '' || (str_starts_with($path, '__groupfolders/') && count(explode('/', $path)) === 2);
+
+		$entry['scan_permissions'] ??= $entry['permissions'];
```

Comment:
You will get a warning if `$entry['permissions']` is not defined, and this is changing the logic of the function which was only doing that if isRoot is true before?